content/articles/2019-04-dbplyr-1-4-0.Rmarkdown

```diff
@@ -0,0 +1,185 @@
+
+You can install the released version from CRAN:
+
+```{r, eval = FALSE}
+install.packages("dbplyr")
+```
+
+While there are a large number of small improvements and bug fixes (see the full full details in the [changelog](https://dbplyr.tidyverse.org/news/index.html)), the biggest improvements in this release are related to SQL generation. In this blog post, I'll show you how find out what dplyr is doing behind the scenes, discuss some major simplifications to the SQL that dbplyr generates, and then show off a few of the new function translations.
+
+To get started, I'll load dplyr and dbplyr:
+
+```{r setup, message = FALSE}
+library(dplyr)
+library(dbplyr)
+```
+
+## SQL simulation
+
+This version of dbplyr substantially improves the tools that allow you to see what SQL dbplyr generates without needing to connect to a live database. You won't generally need these tools for real data, but they're very useful for blog posts (like this one!), for for generating reprexes (as discussed in the new [`vignette("reprex")`](https://dbplyr.tidyverse.org/articles/reprex.html)), and dbplyr's internal tests. 
+
+The basic idea is that you can create a "lazy" frame:
+
+```{r}
+df1 <- lazy_frame(x = 1:3, y = 3:1)
+```
+
+I call this tibble lazy because dplyr operations don't do any work; instead they just record the action so it can later be turned in to a SQL query. You can see this query by printing the lazy tibble:
+
+```{r}
+df1
+```
+
+This is most useful when you add on a few steps:
+
+```{r}
+df1 %>% 
+  group_by(x) %>% 
+  summarise(z = mean(y))
+```
+
+(This example also highlights another small improvement to dbplyr: it'll only warn you about missing values being automatically removed in the database once per session.)
+
+Importantly, `lazy_frame()` has a `con` argument that allows you to specify which database should be used for the translation. This makes it easier to see the differences in SQL generation between databases:
+
+```{r}
+df1 %>% 
+  mutate(z = paste("item:", x)) %>% 
+  head(5)
+
+df2 <- lazy_frame(x = 1, y = 2, con = simulate_mssql())
+df2 %>% 
+  mutate(z = paste("item:", x)) %>% 
+  head(5)
+```
+
+When you don't specify a connection, dplyr uses its standard translation, which tries to follow the ANSI SQL standard as closely as possible.
+
+## Simpler SQL
+
+Two improvements have considerably reduced the number of subqueries that dbplyr needs:
+
+*   Joins, semi joins, and set operations no longer add additional unneeded
+    subqueries and now generate the minimum set:
+
+    ```{r}
+    df1 <- lazy_frame(x = 1, y = 2, a = 2)
+    df2 <- lazy_frame(x = 1, y = 2, b = 2)
+    
+    union(df1, df2)
+    
+    left_join(df1, df2, by = c("x", "y"))
+    
+    semi_join(df1, df2, by = c("x", "y"))
+    ```
+
+*   Many sequences of `mutate()`, `select()`, `rename()`, and `transmute()`
+    steps are be collapsed into a single query: 
```

Comment:
steps _are collapsed_? or steps _are being collapsed_? Currently just _are be collapsed_